metron-interface/metron-alerts/src/app/service/update.service.ts

```diff
@@ -38,22 +41,27 @@ export class UpdateService {
 
   constructor(private http: Http) { }
 
-  public patch(patchRequest: PatchRequest): Observable<{}> {
+  public patch(patchRequest: PatchRequest, fireChangeListner = true): Observable<{}> {
     let url = '/api/v1/update/patch';
     return this.http.patch(url, patchRequest, new RequestOptions({headers: new Headers(this.defaultHeaders)}))
     .catch(HttpUtil.handleError)
     .map(result => {
-      this.alertChangedSource.next(patchRequest);
+      if (fireChangeListner) {
+        this.alertChangedSource.next(patchRequest);
+      }
       return result;
     });
   }
 
-  public updateAlertState(alerts: Alert[], state: string): Observable<{}> {
+  public updateAlertState(alerts: Alert[], state: string, fireChangeListner = true): Observable<{}> {
```

Comment:
is fireChangeListner used in this function?

Reply:
It is passed to patch function and is used there. This is specifically needed in the case where we select multiple rows on the table to update the status. There are too many unnecessary find one calls fired.

Reply:
I agree with you and I think that's a good approach.  The problem is you're not actually using that variable in this method.  Maybe you meant to pass it to `this.patch(patchRequest)` on line 69?